src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -79,5 +83,197 @@ public string VersionString
                 return _versionString;
             }
         }
+
+        /// <summary>
+        /// Indicates whether the current application is running on the specified platform.
+        /// </summary>
+        /// <param name="platform">Case-insensitive platform name. Examples: Browser, Linux, FreeBSD, Android, iOS, macOS, tvOS, watchOS, Windows.</param>
+        public static bool IsOSPlatform(string platform)
+        {
+            if (platform == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.platform);
+            }
+            if (platform.Length == 0)
+            {
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_EmptyOrNullString, ExceptionArgument.platform);
+            }
+
+#if TARGET_BROWSER
+            return platform.Equals("BROWSER", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_WINDOWS
+            return platform.Equals("WINDOWS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_OSX
+            return platform.Equals("OSX", StringComparison.OrdinalIgnoreCase) || platform.Equals("MACOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_UNIX
+            return platform.Equals(s_osPlatformName ??= Interop.Sys.GetUnixName(), StringComparison.OrdinalIgnoreCase);
```

Comment:
`s_osPlatformName` is the only static in this type. Instead of manual lazy initialization, you can initialize it directly and depend on the lazy execution of static constructors to take care of the rest.

Reply:
good point, it's a very cheap method (returns a const) so I've moved it to static field initializer